jsconcat.php

```diff
@@ -31,8 +31,6 @@ function __construct( $scripts ) {
 			}
 			unset( $this->$key );
 		}
-
-		parent::__construct();
```

Comment:
That definitely would do the trick, but the parent constructor would [still register an extra action](https://github.com/WordPress/WordPress/blob/dd6da701b286579819cd6aa518aa2d7018efd759/wp-includes/class.wp-scripts.php#L145) which does not feel right, especially as that action would trigger a method which does nothing.

> That feels like the more "appropriate" approach than not calling the parent constructor.

I would usually agree, but the way the `WPcom_JS_Concat` class is implemented makes me think otherwise. It is designed to replace an, [usually](https://github.com/Automattic/nginx-http-concat/commit/dac2aa0ad3b65c041e7f393a4c63e447574c143a), already existing object and override some of it's methods, while keeping the [original object's properties in place](https://github.com/Automattic/nginx-http-concat/blob/master/jsconcat.php#L25,L34).

In an ideal world, the `WPcom_JS_Concat` class would be used for initialising the global `$wp_scripts` variable, triggering all the actions and getting all the properties set directly.

I really wonder whether there was a good reason for hooking the `js_concat_init` fuction to `init` at priority 10, instead of trying to be called before the global object is created. Brief testing shows that priority set to 0 would do the trick, but it might not be as easy as that.

So, in summary, if we believe that the parent's constructor should be called, then we should revisit the whole implementation, which is not something I wanted to perform in this PR, but am happy to dive into it.